### PR TITLE
Feat(dsheetpiling): Update of the geolib dependencies to run the latest d-sheetpiling console.

### PR DESCRIPTION
**Pull request by Carsopre**

### CODE REVIEW

**Comment by Carsopre** [comment]
This PR is meant to be done once it is decided to 'officially' publish it. In the meantime corrections and separate (related) work can be done using this PR / branch as a base.

**Comment by EleniSmyrniou** [comment]
> This PR is meant to be done once it is decided to 'officially' publish it. In the meantime corrections and separate (related) work can be done using this PR / branch as a base.

Is this because we cannot release a geolib that corresponds to an unrealised version of dsheetpiling?

File: geolib/models/dsheetpiling/internal.py [hunk 1]
```diff
@@ -24,6 +22,7 @@
     DSerieVersion,
 )
 from geolib.utils import make_newline_validator
+from pydantic import confloat, conint, conlist, constr
```
Comment (EleniSmyrniou) [inline geolib/models/dsheetpiling/internal.py:25]: Why did you move external packages a few lines further down the code?

File: geolib/models/dsheetpiling/templates/input.shi.j2 [hunk 2]
```diff
@@ -598,11 +599,11 @@ GeometryDeltaPassiveSurfaceLevel=0.00
 GeometryDeltaPassivePhreaticLine=0.00
 GeometryDeltaActivePhreaticLine=0.00
 OverallStabilityFactorDrivingMoment=1.000
-OverallStabilityFactorCohesion=1.250
-OverallStabilityFactorTgPhi=1.250
+OverallStabilityFactorCohesion=1.000
```
Comment (EleniSmyrniou) [inline geolib/models/dsheetpiling/templates/input.shi.j2:602]: I am guessing that all these factors are changed based on what the DSeries has as default?

File: tests/models/dsheetpiling/test_dsheetpiling_acceptance.py [hunk 3]
```diff
@@ -92,12 +89,14 @@
     Soil,
     SoilType,
 )
+from pydantic.color import Color
+from teamcity import is_running_under_teamcity
```
Comment (EleniSmyrniou) [inline tests/models/dsheetpiling/test_dsheetpiling_acceptance.py:93]: If you removed all uses of this package then delete it

File: tests/models/dsheetpiling/test_dsheetpiling_acceptance.py [hunk 4]
```diff
@@ -92,12 +89,14 @@
...
 from tests.utils import TestUtils, only_teamcity
 
 
 class TestDsheetPilingAcceptance:
+    # @only_teamcity
```
Comment (EleniSmyrniou) [inline tests/models/dsheetpiling/test_dsheetpiling_acceptance.py:98]: Maybe we also delete the comments

File: tests/models/dsheetpiling/test_dsheetpiling_model.py [hunk 5]
```diff
@@ -171,8 +170,22 @@ def test_given_parsed_input_when_serialize_then_same_content(self, filename: Pat
...
                 errors.append(f"Key {ds_key} not serialized!")
                 continue
-            if not (ds_value == output_datastructure[ds_key]):
-                logging.warning(f"UNEQUAL: {ds_value} != {output_datastructure[ds_key]}")
+            od_value = output_datastructure[ds_key]
+            if not (ds_value == od_value):
+                logging.warning(f"UNEQUAL: {ds_value} != {od_value}")
+                try:
+                    # try getting better description of the problem
+                    od_dict = od_value.dict()
```
Comment (EleniSmyrniou) [inline tests/models/dsheetpiling/test_dsheetpiling_model.py:178]: nice

File: tests/models/dsheetpiling/test_dsheetpiling_model.py [hunk 6]
```diff
@@ -229,21 +242,36 @@ def test_execute_console_successfully(self):
...
+        df.serialize(serialized_input_test_file)
+        assert serialized_input_test_file.is_file()
+        df.filename = serialized_input_test_file
         df.execute()
 
         # 3. Verify return code of 0 (indicates succesfull run)
         assert df.datastructure
 
+    @pytest.mark.acceptance
+    @only_teamcity
```
Comment (EleniSmyrniou) [inline tests/models/dsheetpiling/test_dsheetpiling_model.py:261]: Why does this test runs only under teamcity?

File: tests/models/dsheetpiling/test_dsheetpiling_regression_suite.py [hunk 7]
```diff
@@ -14,6 +13,27 @@ class TestDSheetPilingRegressionSuite:
     )
     input_testdata_ids = [str(input_file.stem) for input_file in input_testdata]
 
+    # @pytest.mark.parametrize(
```
Comment (EleniSmyrniou) [inline tests/models/dsheetpiling/test_dsheetpiling_regression_suite.py:16]: Maybe just remove

File: tests/models/dsheetpiling/test_dsheetpiling_model.py (hunk 5 above)
Comment (EleniSmyrniou) [reply tests/models/dsheetpiling/test_dsheetpiling_model.py:178]: Another suggestion for this test(test_import_output) is to run this for all the .shi files in the benchmark directory as an extra test

**Review by EleniSmyrniou — COMMENTED** [review]
Just reviewed your PR

File: geolib/models/dsheetpiling/internal.py (hunk 1 above)
Comment (Carsopre) [reply geolib/models/dsheetpiling/internal.py:25]: This is done automatically by isort, which for some reason was not really working before but it should be an automatic step for everyone.
In my case it gets run by VSCode each time I save a file locally.

File: geolib/models/dsheetpiling/templates/input.shi.j2 (hunk 2 above)
Comment (Carsopre) [reply geolib/models/dsheetpiling/templates/input.shi.j2:602]: Well, that's my assumption based on the files I got by Walter. With these values the whole test bench goes green, so I get the feeling it was good to change them.But perhaps we can have a look at them together.

File: tests/models/dsheetpiling/test_dsheetpiling_model.py (hunk 6 above)
Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_model.py:261]: I believe this is to avoid them running in GitHub. I will have a look either way.

File: tests/models/dsheetpiling/test_dsheetpiling_acceptance.py (hunk 3 above)
Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_acceptance.py:93]: I will check it again 👍

Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_acceptance.py:93]: I removed all unusued references.

File: tests/models/dsheetpiling/test_dsheetpiling_acceptance.py (hunk 4 above)
Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_acceptance.py:98]: Not sure why you get to see this comment because in my latest revision it's actually not commented out.

File: tests/models/dsheetpiling/test_dsheetpiling_model.py (hunk 5 above)
Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_model.py:178]: I believe there's already a test for this.

File: tests/models/dsheetpiling/test_dsheetpiling_model.py (hunk 6 above)
Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_model.py:261]: Confirmed. Because they are long time runs is better to avoid running them in GitHub.

File: tests/models/dsheetpiling/test_dsheetpiling_regression_suite.py (hunk 7 above)
Comment (Carsopre) [reply tests/models/dsheetpiling/test_dsheetpiling_regression_suite.py:16]: Same as before, I do not see this being commented out in my local copy 😞
